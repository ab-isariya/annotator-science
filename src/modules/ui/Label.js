import React from 'react';
import PropTypes from 'prop-types';

/**
 * Component to display label elements inside Cards
 */
const Label = ({primary, secondary, small, children, className}) => {
  // font-size: 14px, line-height: 20px;
  const defaultSizeLineHeight = 'text-sm leading-5';
  // small size, font-size: 12px, line-height: 16px;
  const smallSizeLineHeight = 'text-xs leading-4';

  let classes = `block font-inter uppercase ${className}`;
  // Add text color
  if (primary) {
    classes += ' text-grey-900';
  }
  if (secondary) {
    classes += ' text-grey-500';
  }
  // Add font size
  classes += small ? ` ${smallSizeLineHeight} ` : ` ${defaultSizeLineHeight} `;

  return <span className={classes}>{children}</span>;
};

Label.propTypes = {
  /** black font */
  primary: PropTypes.bool,
  /** grey font */
  secondary: PropTypes.bool,
  /** if true 12px/16px, else 14px/20px */
  small: PropTypes.bool,
  //React node children
  children: PropTypes.node,
  // Tailwind Classes
  className: PropTypes.string
};

export default Label;
