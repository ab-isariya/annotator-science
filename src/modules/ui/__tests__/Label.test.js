import renderer from 'react-test-renderer';

import Label from '../Label';

describe('<Label/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<Label />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with a custom classname', () => {
    const comp = renderer.create(<Label className="testClass" />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });

  it('Renders with with the primary styles', () => {
    const comp = renderer.create(<Label primary />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/text-grey-900/i);
  });

  it('Renders with the secondary styles', () => {
    const comp = renderer.create(<Label secondary />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/text-grey-500/i);
  });

  it('Renders with default font size and line height', () => {
    const comp = renderer.create(<Label />).toJSON();

    expect(comp.props.className).toMatch(/text-sm leading-5/i);
  });

  it('Renders with small size styles', () => {
    const comp = renderer.create(<Label small />).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/text-xs leading-4/i);
  });

  it("Renders it's children", () => {
    const comp = renderer.create(<Label>Hello</Label>).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children).not.toEqual(null);
  });
});
