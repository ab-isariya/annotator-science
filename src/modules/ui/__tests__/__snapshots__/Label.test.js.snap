// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> Renders it's children 1`] = `
<span
  className="block font-inter uppercase undefined text-sm leading-5 "
>
  Hello
</span>
`;

exports[`<Label/> Renders with a custom classname 1`] = `
<span
  className="block font-inter uppercase testClass text-sm leading-5 "
/>
`;

exports[`<Label/> Renders with small size styles 1`] = `
<span
  className="block font-inter uppercase undefined text-xs leading-4 "
/>
`;

exports[`<Label/> Renders with the secondary styles 1`] = `
<span
  className="block font-inter uppercase undefined text-grey-500 text-sm leading-5 "
/>
`;

exports[`<Label/> Renders with with the primary styles 1`] = `
<span
  className="block font-inter uppercase undefined text-grey-900 text-sm leading-5 "
/>
`;

exports[`<Label/> Renders without crashing 1`] = `
<span
  className="block font-inter uppercase undefined text-sm leading-5 "
/>
`;
